src/server/middleware/middleware.go

```diff
@@ -29,3 +29,36 @@ func WithMiddlewares(handler http.Handler, middlewares ...Middleware) http.Handl
 	}
 	return handler
 }
+
+// Skipper defines a function to skip middleware.
+// Returning true skips processing the middleware.
+type Skipper func(*http.Request) bool
```

Comment:
We have two kinds of Middlewares: global middleware and route specific middleware.
For the route specific ones, the skipper isn't neeeded.
For the global ones, is it possible that we implement a function `RegisterGlobalMiddleware(path string, middleware middleware.Middleware)` based on what we have?

Reply:
In my opinion, `Skipper` looks more beautiful than `RegisterGlobalMiddleware `. 

For example, Orm and Transaction middlewares are both only needed by `/v2/*` and `/api/v2.0/*`, then create a skipper to skip other URLs for the middlewares,  it looks very  naturally.  For `RegisterGlobalMiddleware` how to deal with this scenario?  Do we need another middleware to decide to skip Orm or Transaction middleware fo the URL?

And there is a middleware named `Readonly` which will block requests when harbor system in readonly mode but pass the requests if its method is not `POST`, `PUT` or `DELETE`. This scenario can be easily to deal with Skipper, but it is difficult with RegisterGlobalMiddleware.